### PR TITLE
Task6 - Neural Networks

### PR DESCRIPTION
**Pull request by Salvatore112**
# Задача 7. Нейронные сети

* **Дедлайн**: 28.03.2025, 23:59
* **Основной полный балл**: 5
* **Максимум баллов**: 10

## Задача

- [X] Найти данные в виде изображений для задачи классификации. Например, можно взять данные [отсюда](http://ufldl.stanford.edu/housenumbers/).
- [X] Реализовать классы, необходимые для построения сети со следующими слоями
	- FullyConnectedLayer
	- ReluLayer
	- FullyConnectedLayer
- [X] Использовать CrossEntropyLoss и L2-регуляризацию.
- [X] Обучить модель на тренировочных данных, подбирать параметры (особенно learning rate) на валидационной и оценить качество на тестовой. Анализировать графики train/val loss, проверять на каждом шаге корректность вычисления градиентов с помощью разностной оценки.
- [X] * (**+2 балла**) Добавить Batch normalization.
- [X] * (**+2 балла**) В качестве оптимизатор использовать один из: Momentum, RMSprop.
- [X] * (**+1 балл**) Также реализовать оптимизатор Adam.

### CODE REVIEW

**Review by rustam-azimov — APPROVED** [review]
Отлично, 10 баллов
